src/NuGet.Clients/NuGet.CommandLine/Commands/HelpCommandMarkdownTemplateBase.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace NuGet.CommandLine
+{
+    /// <summary>
+    /// Base class for used in the auto-generated Markdown template renderers
+    /// <see cref="HelpCommandMarkdownTemplate"/>
+    /// </summary>
+    /// <remarks>
+    /// Inspired from https://github.com/RazorGenerator/RazorGenerator/blob/master/RazorGenerator.Templating/RazorTemplateBase.cs
+    /// </remarks>
+    public class HelpCommandMarkdownTemplateBase
+    {
+        private System.Text.StringBuilder _generatingEnvironment = new System.Text.StringBuilder();
+        protected System.Text.StringBuilder GenerationEnvironment
+        {
+            get
+            {
+                return this._generatingEnvironment;
+            }
+            set
+            {
+                this._generatingEnvironment = value;
+            }
+        }
+        public virtual void Execute()
```

Comment:
I don't know how the help file gets generated, so maybe this won't work, but this should probably be `abstract` rather than `virtual`.